anchor-tx/src/cas_remote.rs

```diff
@@ -0,0 +1,325 @@
+use std::time::Duration;
+
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use base64::{
+    engine::general_purpose::{STANDARD_NO_PAD as b64_standard, URL_SAFE_NO_PAD as b64},
+    Engine as _,
+};
+use futures::TryStreamExt;
+use multihash_codetable::{Code, MultihashDigest};
+use ring::signature::Ed25519KeyPair;
+use serde::{Deserialize, Serialize};
+use tokio::time::interval;
+use tracing::{debug, warn};
+use uuid::Uuid;
+
+use ceramic_anchor_service::{
+    DetachedTimeEvent, MerkleNode, MerkleNodes, Receipt, TransactionManager,
+};
+use ceramic_car::CarReader;
+use ceramic_core::{
+    cid_from_ed25519_key_pair, did_key_from_ed25519_key_pair, Cid, StreamId, StreamIdType,
+};
+use ceramic_event::unvalidated::Proof;
+
+pub const AGENT_VERSION: &str = concat!("ceramic-one/", env!("CARGO_PKG_VERSION"));
+
+#[derive(Serialize, Deserialize, Debug)]
+struct CasAuthPayload {
+    url: String,
+    nonce: String,
+    digest: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+struct CasAnchorRequest {
+    stream_id: StreamId,
+    cid: String,
+    timestamp: String,
+    ceramic_one_version: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Header {
+    pub alg: String,
+    pub kid: String,
+    pub typ: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Claims {
+    nonce: String,
+    url: String,
+    digest: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+struct AnchorResponse {
+    pub message: String,
+    pub witness_car: Option<String>,
```

Comment:
nit: should this be a byte vector instead of a string?

Reply:
This matches what the CAS responds with, which is a base64-encoded string for the witness CAR file.

Reply:
We added a comment clarifying this:
```
struct AnchorResponse {
    pub message: String,
    /// This field contains the Base64 Standard encoding of the witness CAR file
    pub witness_car: Option<String>,
}
```